### PR TITLE
chore: product refactoring

### PR DESCRIPTION
**Pull request by tobi-or-not-tobi**
- use signals
- use schema's

closes [HRZ-2647](https://spryker.atlassian.net/browse/HRZ-2647)

[HRZ-2647]: https://spryker.atlassian.net/browse/HRZ-2647?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/domain/product/media/src/media.component.ts [hunk 1]
```diff
@@ -67,21 +48,17 @@ export class ProductMediaComponent extends ProductMixin(
...
 
-  protected renderVideo(src: string): TemplateResult | void {
-    return html`<oryx-video url=${src}></oryx-video>`;
-  }
-
   /**
    * Resolves the media set. When there's no mediaSet index provided, the first
    * media set is returned.
    */
   protected getMediaSet(): ProductMediaSet | undefined {
```
Comment (dunqan) [inline libs/domain/product/media/src/media.component.ts]: could be changed to computed

File: libs/domain/product/src/services/product-list.service.ts [hunk 2]
```diff
@@ -8,7 +8,7 @@ export interface ProductListService {
     qualifier: ProductListQualifier
   ): Observable<QueryState<ProductList | undefined>>;
 
-  getSearchParams(qualifier: ProductListQualifier): Record<string, string>;
+  // getSearchParams(qualifier: ProductListQualifier): Record<string, string>;
```
Comment (dunqan) [inline libs/domain/product/src/services/product-list.service.ts]: left over

File: libs/platform/core/src/services/context/context.controller.ts [hunk 3]
```diff
@@ -37,6 +38,10 @@ export class ContextController implements ReactiveController {
       ),
     ]).pipe(
       skip(1),
+      tap(([overrideContext, context]) => {
```
Comment (dunqan) [inline libs/platform/core/src/services/context/context.controller.ts:41]: we should have unit tests for this behavior, also maybe @supproduction can take a look, if we don't break some edge cases (when we already have context provided on the host + override for example)

File: libs/domain/product/card/src/card.component.ts [hunk 4]
```diff
@@ -38,52 +29,43 @@ export class ProductCardComponent extends ProductMixin(
 
   protected context = new ContextController(this);
```
Comment (supproduction) [inline libs/domain/product/card/src/card.component.ts]: Do we need it?

File: libs/domain/product/src/services/product-list.service.ts (hunk 2 above)
Comment (tobi-or-not-tobi) [reply libs/domain/product/src/services/product-list.service.ts]: done

File: libs/domain/product/media/src/media.component.ts (hunk 1 above)
Comment (tobi-or-not-tobi) [reply libs/domain/product/media/src/media.component.ts]: done

**Review by supproduction — DISMISSED** [review]

**Review by dunqan — APPROVED** [review]
